### PR TITLE
Create pids.geoconnex.us User and Contributor documentation

### PR DESCRIPTION
**Pull request by webb-ben**
This PR adds documentation for the Permanent Identifier Service and all of its related dependencies as deployed in the geoconnex.us system.

### CODE REVIEW

File: docs/contributor/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+## Topics
+For specific contributor documentation on a particular software, follow the contributor spepcific documentation:
+- [Sitemap Generator](./sitemap-generator/)
+- [Yourls Action](./yourls-action/)
+
+
+# Prerequisite Knowledge
+
+The Contributor should become proficient with the following programming languages, data models, file formats, toolchains, and last but not least, the Geoconnex Components and related dependent services that comprise the System.
```
Comment (ewiggansLI) [inline docs/contributor/README.md:16]: I don't think you need to say "last but not least" but I also appreciated the brevity

File: docs/contributor/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+## Programming Languages
+
+### [Python](https://python.org)
+What the [Yourls Action](https://github.com/cgs-earth/yourls-action) and [Sitemap Generator](https://github.com/cgs-earth/sitemap-generator) for the Geoconnex Permanent Identifier (PID) Service are written in.
+
+### [PHP](https://www.php.net/)
+What the [PID Service](#pid-server) Geoconnex component is written in.
+
+### [Bash](https://www.gnu.org/software/bash/manual/bash.html)
+Used to create various scripts in the [Geoconnex component](#geoconnex-components) toolchain
```
Comment (ewiggansLI) [inline docs/contributor/README.md:27]: Very nitpicky here but the other lines have periods at the ends; this one doesn't

File: docs/contributor/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+An xml document that contains URLs of a Publisher's http endopoints for which JSON-LD documents can be extracted and aggregated.
+
+### [Sitemap Index](https://www.sitemaps.org/protocol.html#index)
+An xml document that contains URLs of Publisher specific [sitemaps](#Sitemap), 
+
+## Tool(chain)s
+
+[Geoconnex components](#geoconnex-components) are invariably deployed and orchestrated as Docker containers at runtime. 
+
+### [docker](https://docs.docker.com/engine/reference/commandline/cli/)
```
Comment (ewiggansLI) [inline docs/contributor/README.md:50]: Should this have anything under it the other things do

File: docs/contributor/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+[Geoconnex components](#geoconnex-components) are invariably deployed and orchestrated as Docker containers at runtime. 
+
+### [docker](https://docs.docker.com/engine/reference/commandline/cli/) 
+### [docker compose](https://docs.docker.com/compose/)
+*See Also* [Yaml](#yaml), 
+
+## Software Frameworks
+
+[Yourls](https://yourls.org/) is an open source URL shorter which the [PID Server](#pid-server) is built upon and deployed by [Aggregators](https://github.com/internetofwater/harvest.geoconnex.us/blob/main/README.md#persona-aggregator). The data pipeline created by the System which connects [Publishers](https://github.com/internetofwater/harvest.geoconnex.us/blob/main/README.md#persona-publisher) to [Aggregators](https://github.com/internetofwater/harvest.geoconnex.us/blob/main/README.md#persona-aggregator) is performed here.
```
Comment (ewiggansLI) [inline docs/contributor/README.md:56]: Maybe my ignorance, should Aggregators have its own section? or is it an idea/concept generally? I wonder then if it should be capitalized?

File: docs/contributor/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+## System Components 
+
+### Geoconnex Components
+
+#### [PID Server](https://github.com/internetofwater/pids.geoconnex.us) 
+
+Pemanent Identifier (PID) Server is used to provide re-directs to Geoconnex webpages and provides a [Sitemap Index](#sitemap-index) of JSON-LD pages that [Gleaner](https://github.com/gleanerio/gleaner) ingests to geoconnex. It is implemented using cloud native services, leveraging GCP's Cloud Run and Cloud SQL.
```
Comment (ewiggansLI) [inline docs/contributor/README.md:64]: You already defined the abbreviation for PID above so I don't think you need to spell it out here again the same way, but for clarity it also isn't the worst.

Comment (ewiggansLI) [reply docs/contributor/README.md:64]: Also JSON-LD isn't mentioned in file formats above, should it be?

File: docs/contributor/yourls-action/README.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+docker run -d \
+   -p 3306:3306 \
+   --name mysql \
+   -e "MYSQL_ROOT_PASSWORD=amazingpassword" \
+   yourls-mysql
+```
+
+### Create a namespace
+
+Create a namespace filesystem, and a CSV mapping file [links.csv](./links.csv).
```
Comment (ewiggansLI) [inline docs/contributor/yourls-action/README.md:64]: Is it going to be clear to a user what a namespace filesystem is if so disregard but I would be like 'whaaat'

File: docs/contributor/yourls-action/README.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+YOURLS Action uses environment variables to connect to the MySQL database:
+```bash
+export DB_SOCKET_DIR=/var
+export YOURLS_DB_HOST=localhost
+export YOURLS_DB_PASSWORD=amazingpassword
+```
+
+# Running YOURLS Action
+
+To run YOURLS Action, use the CLI:
```
Comment (ewiggansLI) [inline docs/contributor/yourls-action/README.md:92]: Do you need to define CLI somewhere, again, I actually know this one but I'm looking out for the others like me who may not! But maybe it is common enough you don't

File: docs/quickstart.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
+# Quickstart
+
+## About
+
+This document provides a quick set up for an up-to-date version of the Permanent Identifier Service. For a quickstart on YOURLS, see https://yourls.org/docs/guide/install. For a quickstart as a contributor, see [Contributor](./contributor/) for the set up of a development environment.
```
Comment (ewiggansLI) [inline docs/quickstart.md:5]: You refer to PID above so you could probably say that here, unless it has a different meaning here.

**Review by ewiggansLI — COMMENTED** [review]
Hopefully this was helpful from someone with some but not as much working knowledge as you!

File: docs/contributor/README.md (hunk 2 above)
Comment (webb-ben) [reply docs/contributor/README.md:27]: nitpick is why you review :)

File: docs/contributor/README.md (hunk 4 above)
Comment (webb-ben) [reply docs/contributor/README.md:56]: Its a concept in gleaner / harvest.geoconnex.us

File: docs/contributor/README.md (hunk 5 above)
Comment (webb-ben) [reply docs/contributor/README.md:64]: Good catch

File: docs/contributor/yourls-action/README.md (hunk 7 above)
Comment (webb-ben) [reply docs/contributor/yourls-action/README.md:92]: CLI = Command Line Interface, or the Python terminal in ArcPro. Will define above

**Pull request merged**